packages/squiggle-lang/src/cli/makeProgram.ts

```diff
@@ -34,24 +55,36 @@ export const makeProgram = () => {
         output = "RESULT_AND_BINDINGS";
       }
 
-      if (filename && options.eval) {
-        program.error("Only one of filename and eval string should be set.");
-      } else if (filename) {
-        src = fs.readFileSync(filename, "utf-8");
-      } else if (options.eval) {
-        src = options.eval;
-      } else {
-        program.error("One of filename and eval string should be set.");
-      }
+      const src = loadSrc({ filename, inline: options.eval });
 
       const sampleCount = process.env.SAMPLE_COUNT;
 
       run({ src, filename, output, measure: options.time, sampleCount });
     });
 
+  program
+    .command("parse")
+    .arguments("[filename]")
+    .option(
+      "-e, --eval <code>",
+      "parse a given squiggle code string instead of a file"
+    )
+    .action((filename, options) => {
+      const src = loadSrc({ filename, inline: options.eval });
+
+      const parseResult = parse(src);
+      if (parseResult.ok) {
+        console.log(
+          util.inspect(parseResult.value, { depth: Infinity, colors: true })
```

Comment:
This prints a huge JSON, so it might be not exactly what was meant in #708. But it'll be easy to add `--quiet` option to this command if we ever need it.

For now, I needed it for prettier plugin debugging.

Reply:
In my story, I was imagining that this would basically just verify that the compilation was successful, not that it would be showed in the console. That said, I'm happy with this option.